.github/workflows/gradle.yml

```diff
@@ -10,48 +10,58 @@ jobs:
   build:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
       - name: Set up JDK
-        uses: actions/setup-java@v3
+        uses: actions/setup-java@v4
         with:
           distribution: temurin
           java-version: 17
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@v3
+        with:
+          # Publish a Gradle build scan so you can see all the build logs,
+          # a complete task timeline, test outputs, and the resolved dependencies of your build.
+          build-scan-publish: true
+          build-scan-terms-of-use-url: "https://gradle.com/terms-of-service"
```

Comment:
We have a Develocity (former Gradle Enterprise) instance (configured in settings.gradle. With that, the only item needed here is develocity-access-key: ${{ secrets.GRADLE_ENTERPRISE_ACCESS_KEY }}.